wasm.md

```diff
@@ -1011,6 +1016,52 @@ The maximum of table size is 2^32 bytes.
     rule #maxTableSize()  => 4294967296
 ```
 
+### Data Segments
+
+Memories can be initialized with data, specified as a list of bytes together with an offset.
+The `data` initializer simply puts these bytes into the specified memory, starting at the offset.
+
+```k
+    syntax Instr ::= Data
+    syntax Data ::= "(" "data" MemId DataOffset DataStrings ")"
+                  | "(" "data"       DataOffset DataStrings ")"
+                  |     "data" "{" MemId DataStrings "}"
+ // ----------------------------------------------------
+    // Default to memory 0.
+    rule <k> ( data       OFFSET STRINGS ) => ( data 0 OFFSET STRINGS ) ... </k>
+    rule <k> ( data MEMID    IS:Instr STRINGS ) => IS ~> data { MEMID STRINGS } ... </k>
+    rule <k> ( data MEMID (offset IS) STRINGS ) => IS ~> data { MEMID STRINGS } ... </k>
+
+    // For now, deal only with memory 0.
+    rule <k> data { MEMIDX STRING } => . ... </k>
+         <valstack> < i32 > OFFSET : STACK => STACK </valstack>
+         <memIndices> MEMIDX |-> ADDR </memIndices>
+         <memInst>
+           <mAddr> ADDR </mAddr>
+           <mdata>   DATA
+                  => #clearRange(DATA, OFFSET, OFFSET +Int #dataStringsLength(STRING) -Int 1) [ OFFSET := #dataStrings2int(STRING)]
+           </mdata>
+           ...
+         </memInst>
+
+    syntax Int ::= #lengthDataPages        ( Data ) [function]
+                 | Int "/ceilInt" Int          [function]
+ // -----------------------------------------------------
```

Comment:
Formatting here is kinda weird.

Reply:
Hopefully that's better. Also got rid of `#if` in favor of `requires`